feature: output external log files with colored

YusukeHosonuma opened the pull request:
resolved: #140 #141 

## Output example
![image](https://user-images.githubusercontent.com/2990285/104607621-5d126f00-56c4-11eb-817d-a922ce2ed3de.png)

## Combine extension
![image](https://user-images.githubusercontent.com/2990285/104622606-d44fff00-56d4-11eb-86a6-a55a4f855a7f.png)

YusukeHosonuma committed:
feature: output external log files with colored

YusukeHosonuma committed:
update .podspec and others

YusukeHosonuma committed:
feature: output to external log files on Combine extension

YusukeHosonuma committed:
make public the ColorTheme

YusukeHosonuma committed:
specify platforms

YusukeHosonuma committed:
update README

YusukeHosonuma committed:
update CI workflow

YusukeHosonuma committed:
fix compile error on Swift 5.1

YusukeHosonuma committed:
update workflows

YusukeHosonuma committed:
update README

YusukeHosonuma committed:
Combine extension tests are running on available environments only

sahara-ooga commented on Sources/Core/PrettyDescriber.swift:
```diff
@@ -6,10 +6,12 @@
 // Copyright (c) 2020 Yusuke Hosonuma.
 //
 
+import ColorizeSwift
```
```suggestion

```

You don't have to import `ColorizeSwift` in this file.

In other words, you can build without this `import` statements.

Commit:
Update Data.withUnsafeBytes method to new version

# Reference

[withUnsafeBytes(_:) | Apple Developer Documentation](https://developer.apple.com/documentation/foundation/data/1780450-withunsafebytes) (deprecated)

[withUnsafeBytes(_:) | Apple Developer Documentation](https://developer.apple.com/documentation/swift/array/2635991-withunsafebytes)

[UnsafeRawPointer | Apple Developer Documentation](https://developer.apple.com/documentation/swift/unsaferawpointer)

[assumingMemoryBound(to:) | Apple Developer Documentation](https://developer.apple.com/documentation/swift/unsaferawpointer/2431721-assumingmemorybound)

[withUnsafeBytes deprecated, is thi… | Apple Developer Forums](https://developer.apple.com/forums/thread/116309)

[Swift.org - UnsafeRawPointer Migration](https://swift.org/migration-guide-swift3/se-0107-migrate.html)

[素晴らしいSwiftのポインタ型の解説 - Qiita](https://qiita.com/omochimetaru/items/c95e0d36ae7f1b1a9052) (In Japanese only)

Commit:
add requirements for the example project

Commit:
Add a comment for newly added argument `colored` of `print` method of Pretty class

Commit:
Separete argument names into internal and external argument names

In CombineExtension.swift, arguments named "to" cause misleading codes like `var out = to`,  `_out(value, to: to)` and `if to is StandardOutput`.
Such confusion can be resolved by separating the external and internal argument names.

sahara-ooga reviewed (approved):
These changes are very nice additions.
There are a few things that bothered me, so I wrote some [suggestions for changes](https://github.com/YusukeHosonuma/SwiftPrettyPrint/pull/143#discussion_r564347870) and pull requests ( #149 #150 #151 #152 ). 
I hope you will find them useful.

YusukeHosonuma committed:
Merge pull request #152 from YusukeHosonuma/feature/CombineExtension/Publisher/prettyPrint/separete-argument-names-into-internal-and-external-argument-names

Separete argument names into internal and external argument names

YusukeHosonuma committed:
Merge pull request #151 from YusukeHosonuma/feature/Pretty/print/add-a-comment

Add a comment for newly added argument

YusukeHosonuma committed:
Merge pull request #150 from YusukeHosonuma/feature/Example/README.md/Requrements

Add requirements for the example project In README.md

YusukeHosonuma committed:
Merge pull request #149 from YusukeHosonuma/feature/log-update-withUnsafeBytes

Update Data.withUnsafeBytes method to new version

YusukeHosonuma committed:
Merge pull request #146 from YusukeHosonuma/support/platforms

Clarification of supported versions

YusukeHosonuma committed:
Update Sources/Core/PrettyDescriber.swift

Co-authored-by: sahara-ooga <performa532@gmail.com>

YusukeHosonuma commented:
@sahara-ooga 
Thanks a lot for review and many suggestion changes 👍
